service/history/workflow/mutable_state_impl.go

```diff
@@ -2144,9 +2192,13 @@ func (ms *MutableStateImpl) ReplicateActivityTaskScheduledEvent(
 		}
 	}
 
+	if prev, existed := ms.pendingActivityInfoIDs[ai.ScheduledEventId]; existed {
```

Comment:
nit: for xxxScheduled & xxx Initiated event & timerStarted, pending maps won't contain the activity or signal or cancelInfo we are trying to add.